test/jdk/java/net/httpclient/http2/server/OutgoingPushPromise.java

```diff
@@ -45,4 +49,17 @@ public OutgoingPushPromise(int parentStream,
         this.parentStream = parentStream;
     }
 
+    public void addContinuation(ContinuationFrame cf) {
+        if (continuations == null)
+            continuations = new LinkedList<>();
+        continuations.add(cf);
+    }
```

Comment:
I would suggest adding a constructor that takes a list of continuations/frames instead. This way you could have an immutable list (use List.of/List.copyOf)

Reply:
Ah, good suggestion!